### PR TITLE
[P0] Add Fluent theme pack

### PR DESCRIPTION
**Pull request by compulim**
### Added

-  Added `sendBoxMiddleware` and `sendBoxToolbarMiddleware`, by [@compulim](https://github.com/compulim), in PR [#5120](https://github.com/microsoft/BotFramework-WebChat/pull/5120)
-  Added `botframework-webchat-fluent-theme` package for applying Fluent UI theme to Web Chat, by [@compulim](https://github.com/compulim), in PR [#5120](https://github.com/microsoft/BotFramework-WebChat/pull/5120)
-  Added `<ThemeProvider>` component to apply theme pack to Web Chat, by [@compulim](https://github.com/compulim), in PR [#5120](https://github.com/microsoft/BotFramework-WebChat/pull/5120)

## Description

Apply a preset of middleware and style options in the following order: props > theme > default.

```tsx
import { createRoot } from 'react-dom/client';
import ReactWebChat, { Components } from 'botframework-webchat';

const { ThemeProvider } = Components;

createRoot(
  <ThemeProvider styleOptions={{ bubbleBackground: '#fee' }}>
    <ReactWebChat {...props} />
  </ThemeProvider>,
  document.getElementByTagNames('main')[0]
);
```

## Design

- `<ThemeProvider>` is outside of Web Chat
   - `<ThemeProvider>` must not use any Web Chat hooks that requires `<Composer>`
- `<ThemeProvider>` can be applied at page-level, or just outer of `<ReactWebChat>`
- `<ThemeProvider>` can be nested/cascaded
- Whitelisted props from `<Composer>` only
   - Limit what we allow/expose, simpler logics and a "cleanup" of what we have today
   - Some props have different mixing/merging logic than other props
      - Middleware merge as array
      - Style options merge as object
- Need to merge/expose at `component` package because it is where the default middleware is introduced
    - We need to inject the theme in this order: props > theme > default
    - We need to inject at a package that contains both props and default, i.e. the `component` package
    - Hope to see if we can do it at `api` package, because it feels more neutral, P1/P2
       - `component` package means React HTML 
       - `api` package means React, that could mean React HTML or React Native
       - Providing a theme, is a neutral task (the themed content is unlikely to be neutral though)
- Added `sendBoxMiddleware` and `sendBoxToolbarMiddleware`
- Added `botframework-webchat-fluent-theme` package and wired visual regression tests

## Specific Changes

- Added `botframework-webchat-fluent-theme` package and wired visual regression tests
- Added `sendBoxMiddleware` and `sendBoxToolbarMiddleware`
   - `<BasicSendBox>` will be the default of `sendBoxMiddleware`
   - `<UploadButton>` will be the default of `sendBoxToolbarMiddleware`
- Added `<ThemeProvider>` in `botframework-webchat-component` so it will merge props in `<Component.Composer>`
   - Later, we should move `<ThemeProvider>` to API to broaden theme support

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [x] I have updated documentation

## Review Checklist

> This section is for contributors to review your work.

-  [x] ~Accessibility reviewed (tab order, content readability, alt text, color contrast)~
-  [x] ~Browser and platform compatibilities reviewed~
-  [x] ~CSS styles reviewed (minimal rules, no `z-index`)~
-  [x] ~Documents reviewed (docs, samples, live demo)~
-  [x] ~Internationalization reviewed (strings, unit formatting)~
-  [x] `package.json` and `package-lock.json` reviewed
-  [x] ~Security reviewed (no data URIs, check for nonce leak)~
-  [x] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: __tests__/hooks/useGroupTimestamp.js [hunk 1]
```diff
@@ -19,18 +19,6 @@ test('getter should return group timestamp set in styleOptions', async () => {
   expect(groupTimestamp).toMatchInlineSnapshot(`1000`);
 });
 
-test('getter should return group timestamp set in props (deprecated)', async () => {
```
Comment (compulim) [inline __tests__/hooks/useGroupTimestamp.js:22]: Removed deprecated code, so removing corresponding tests.

File: __tests__/hooks/useTimeoutForSend.js [hunk 2]
```diff
@@ -19,21 +19,8 @@ test('getter should return timeout for sending activity', async () => {
   expect(timeoutForSend).toMatchInlineSnapshot(`1000`);
 });
 
-test('getter should return timeout for sending activity if set in props', async () => {
```
Comment (compulim) [inline __tests__/hooks/useTimeoutForSend.js:22]: Removed deprecated code, so removing corresponding tests.

File: packages/api/src/hooks/Composer.tsx [hunk 3]
```diff
@@ -283,18 +264,14 @@ const ComposerCore = ({
   const patchedDir = useMemo(() => (dir === 'ltr' || dir === 'rtl' ? dir : 'auto'), [dir]);
   const patchedGrammars = useMemo(() => grammars || [], [grammars]);
   const patchedStyleOptions = useMemo(
-    () => normalizeStyleOptions(patchStyleOptionsFromDeprecatedProps(styleOptions, { groupTimestamp, sendTimeout })),
-    [groupTimestamp, sendTimeout, styleOptions]
```
Comment (compulim) [inline packages/api/src/hooks/Composer.tsx:287]: Remove deprecated code.

File: packages/component/src/SendBox/BasicSendBox.tsx [hunk 4]
```diff
@@ -97,14 +95,6 @@ const BasicSendBox: FC<BasicSendBoxProps> = ({ className }) => {
   );
 };
```
Comment (compulim) [inline packages/component/src/SendBox/BasicSendBox.tsx:96]: defaultProps are deprecated by newer version of React.

**Review by tdurnford — APPROVED** [review]

**Pull request merged**